### PR TITLE
FOGL-1917: Support added for child categories for the South pages

### PR DESCRIPTION
**Pull request by mshariq-nerd**
Support added to the south page modal to able to access and edit the content of the child category if that is defined for a south service instance.

Check attached screenshots for better understanding

hidden_advance_config

child-config

**EDIT:** ~This PR is need to be tested with https://github.com/foglamp/FogLAMP/pull/1189 for proper display name.~  `FogLAMP/pull/1189` merged into develop!

### CODE REVIEW

**Comment by praveen-garg** [comment]
@mshariq-nerd  please rebase, resolve the conflicts and retest with develop.

File: src/app/components/core/south/south-service-modal/south-service-modal.component.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import { Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewChild } from '@angular/core';
 import { FormControl } from '@angular/forms';
-import { isEmpty } from 'lodash';
+import { isEmpty, some } from 'lodash';
```
Comment (monika-Dianomic) [inline src/app/components/core/south/south-service-modal/south-service-modal.component.ts]: `some` is not required here.

File: src/app/components/core/configuration-manager/config-children/config-children.component.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+                    <textarea rows="3" *ngIf="getConfigAttributeType(configItem.type) === 'LONG_TEXT'" class="textarea is-fullwidth is-small"
+                      name="{{configItem.key}}" [ngModel]="setConfigValue(configItem)" [value]='setConfigValue(configItem)'
+                      [attr.maxLength]="configItem?.length" [jsonValue]="setConfigValue(configItem)" required>{{configItem?.value != undefined ? configItem?.value : configItem?.default}}</textarea>
+
+                    <div *ngIf="getConfigAttributeType(configItem.type) == 'ENUMERATION'" class="select is-small">
+                      <select name='{{configItem.key}}' [ngModel]="setConfigValue(configItem)">
+                        <option *ngFor="let configOption of configItem.options" [value]='configOption'>{{configOption}}</option>
+                      </select>
+                    </div>
```
Comment (monika-Dianomic) [inline src/app/components/core/configuration-manager/config-children/config-children.component.html]: Required to handle `SCRIPT` type configItem as it is in now.

Comment (praveen-garg) [reply src/app/components/core/configuration-manager/config-children/config-children.component.html]: No actually, There is no use case yet to handle script type in this component.

**Comment by billehunt** [comment]
looks awesome. thanks!

**Pull request merged**